### ISSUE TITLE
Overriding components as part of data cascade

### ISSUE THREAD

**Issue by pauleveritt** [11ty/eleventy-plugin-webc#37]
*tl;dr Allow customization of components in certain parts of the site.*

This is very much a blue-sky feature. Gatsby has a feature called [theme shadowing](https://www.gatsbyjs.com/docs/how-to/plugins-and-themes/shadowing/). The idea: someone can consume a package of components, but replace just one of them without forking. I use it in a multi-site project.

The WebC plugin currently disallows "registration" of more-than-one implementation of a component name. The data cascade lets you override a variable as you get more specific. Imagine if you could, in a directory or template, say "Look here for local component definitions."

I realize this indirection would be painful for several reasons so feel free to immediately close it with a "put down the crack pipe" comment.

**Comment by zachleat** [comment]
I think there is something here! Folks should upvote this if they want it too!
